tactics/hints.ml

```diff
@@ -916,19 +902,25 @@ let fresh_global_or_constr env sigma poly cr =
       (c, Univ.ContextSet.empty)
     end
 
-let make_resolves env sigma flags info ~check ~poly ?name cr =
+let make_resolves env sigma (eapply, hnf) info ~check ~poly ?name cr =
   let c, ctx = fresh_global_or_constr env sigma poly cr in
   let cty = Retyping.get_type_of env sigma c in
   let try_apply f =
-    try Some (f (c, cty, ctx)) with Failure _ -> None in
+    try
+      let (_, hint) as ans = f (c, cty, ctx) in
+      match hint.code.obj with
+      | ERes_pf _ -> if not eapply then None else Some ans
```

Comment:
```suggestion
      | ERes_pf _ when not eapply -> None
```
?

Reply:
I hate when clauses, they break linearity and are easy to mess up. Please don't force me to write one.

Reply:
ok